deployment/promotion/promote.py

```diff
@@ -35,13 +37,19 @@
 
 env_vars = ['VERTA_SOURCE_MODEL_VERSION_ID', 'VERTA_SOURCE_HOST', 'VERTA_SOURCE_EMAIL', 'VERTA_SOURCE_DEV_KEY',
             'VERTA_SOURCE_WORKSPACE', 'VERTA_DEST_HOST', 'VERTA_DEST_EMAIL', 'VERTA_DEST_DEV_KEY',
-            'VERTA_DEST_WORKSPACE']
+            'VERTA_DEST_WORKSPACE', 'VERTA_DEST_REGISTERED_MODEL_ID']
 params = {}
 
+create_new_registered_model = False
+
 for param_name in env_vars:
     param = os.environ.get(param_name)
     if not param:
-        raise Exception("Missing environment variable %s", param_name)
+        # 'VERTA_DEST_REGISTERED_MODEL_ID' is optional; don't exit
+        if param_name != 'VERTA_DEST_REGISTERED_MODEL_ID':
+            raise Exception("Missing environment variable %s", param_name)
+        else:
+            create_new_registered_model = True
     params[param_name] = param
```

Comment:
This loop feels confusing to me since it's wearing requived and optional env vars, plus setting `create_new_registered_model = True` is buried inside two `if`s which has my mind in a pretzel.

What would you think about

- putting `VERTA_DEST_REGISTERED_MODEL_ID` in a separate `optional_env_vars` list and having a second `for` loop for that
- instead of having `create_new_registered_model` at all, use `"VERTA_DEST_REGISTERED_MODEL_ID" not in params`